sycl/plugins/level_zero/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+# PI Level0 plugin library
+
+message(STATUS "Download Level Zero loader and headers from github.com")
+if(MSVC)
+  set(L0_LIBRARY
+      "${LLVM_LIBRARY_OUTPUT_INTDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}ze_loader${CMAKE_STATIC_LIBRARY_SUFFIX}")
+else()
+  set(L0_LIBRARY
+      "${LLVM_LIBRARY_OUTPUT_INTDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}ze_loader${CMAKE_SHARED_LIBRARY_SUFFIX}")
+endif()
+if (CMAKE_C_COMPILER)
+    list(APPEND AUX_CMAKE_FLAGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER})
+endif()
+if (CMAKE_CXX_COMPILER)
+    list(APPEND AUX_CMAKE_FLAGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER})
+endif()
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/l0_loader_build)
+ExternalProject_Add(l0-loader
+  GIT_REPOSITORY    https://github.com/oneapi-src/level-zero.git
+  GIT_TAG           origin/master
+  SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/Level0/l0_loader"
+  BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/l0_loader_build"
+  INSTALL_DIR       "${CMAKE_CURRENT_BINARY_DIR}/l0_loader_install"
+  CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+             -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+             -DOpenCL_INCLUDE_DIR=${OpenCL_INCLUDE_DIRS}
+             -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+             -DCMAKE_INSTALL_LIBDIR:PATH=lib${LLVM_LIBDIR_SUFFIX}
+             ${AUX_CMAKE_FLAGS}
+  STEP_TARGETS      configure,build,install
+  DEPENDS           ocl-headers
+  BUILD_BYPRODUCTS ${L0_LIBRARY}
+)
+ExternalProject_Add_Step(l0-loader llvminstall
```

Comment:
It looks like l0-loader is not added to deploy-sycl-toolchain. So it will not be available after recommended build process. So if we build compiler and runtime on machine with GPU it might be non-functional due to missed dependency library (l0-loader)

Reply:
@vladimirlaz I have a question, do we deploy OpenCL.so? I tried to do ninja install and don't see ICD in the installation directory. How this works for OpenCL?

Reply:
@againull you need to download it on build time (rather than using pre-installed) and specify a special flag on cmake configuration. See https://github.com/intel/llvm/commit/fd64c8f94fb6217a2d6760bf59596c66da3aeb91#diff-ad9ca624b2a795a5f24145a17c806d1f

Reply:
Added l0-loader to deploy-sycl-toolchain. One question: how the scenario described by Vladimir will work for OpenCL if ICD is added to deploy-sycl-toolchain only under special option?